### PR TITLE
Add ability to exclude widget from being wrapped in <Card />

### PR DESCRIPTION
**Pull request by Guap6512**
Adds ability to exclude custom widgets from being wrapped in <Card /> component

### CODE REVIEW

File: src/components/Widget/Widget.tsx [hunk 1]
```diff
@@ -44,7 +45,9 @@ export const Widget: FunctionComponent<WidgetProps> = (props) => {
     if (!props.showWidget) {
         return null
     }
-    if (props.meta.type === WidgetTypes.AssocListPopup || props.meta.type === WidgetTypes.PickListPopup) {
+    if (props.meta.type === WidgetTypes.AssocListPopup
+        || props.meta.type === WidgetTypes.PickListPopup
+        || props?.skipWrappingWidgets.includes(props.meta.type)) {
```
Comment (DRITE) [inline src/components/Widget/Widget.tsx:50]: I guess here should be something like this
```
|| props.skipWrappingWidgets?.includes(props.meta.type)) {
```

**Comment by DRITE** [comment]
@Guap6512 there are  some problems now:
lint error -  https://github.com/tesler-platform/tesler-ui/pull/380/checks?check_run_id=944592395#step:4:8
UT error - https://github.com/tesler-platform/tesler-ui/pull/380/checks?check_run_id=944592314#step:4:11

**Comment by Guap6512** [comment]
> @Guap6512 there are some problems now:
> lint error - https://github.com/tesler-platform/tesler-ui/pull/380/checks?check_run_id=944592395#step:4:8
> UT error - https://github.com/tesler-platform/tesler-ui/pull/380/checks?check_run_id=944592314#step:4:11

fixed

File: src/components/Widget/Widget.tsx [hunk 2]
```diff
@@ -33,18 +33,24 @@ interface WidgetProps extends WidgetOwnProps {
...
 
+const SKIP_WRAPPING_WIDGETS: string[] = [WidgetTypes.AssocListPopup, WidgetTypes.PickListPopup]
+
 export const Widget: FunctionComponent<WidgetProps> = (props) => {
     if (!props.showWidget) {
         return null
     }
-    if (props.meta.type === WidgetTypes.AssocListPopup || props.meta.type === WidgetTypes.PickListPopup) {
+
+    const skipCardWrapping = SKIP_WRAPPING_WIDGETS.includes(props.meta.type) && props.skipWrappingWidgets?.includes(props.meta.type)
```
Comment (DRITE) [inline src/components/Widget/Widget.tsx:51]: Why conjunction has appeared here instead of disjunction?

Comment (Guap6512) [reply src/components/Widget/Widget.tsx]: fixed

**Comment by Guap6512** [comment]
> 1. We need a broader approach to customization. If some widget types does not require a `<Card />` wrapper at all the next logical step is that different widget types will require a different `<Card />` implementations, and then we will stuck with over-complicated component props (skipWidget.
> 
> Try this instead:
> 
> ```tsx
> // Keep old behavior backward-compatible and extend it with optional config:
> /**
>  * TSDoc
>  */
> type CustomWidgetDescriptor = CustomWidget | {
>     /**
>      * TSDoc
>      */
>     component: CustomWidget,
> 
>     /**
>      * TSDoc
>      */
>     card?: CustomWidget
> }
> 
> export interface ViewProps {
>     customWidgets?: Record<string, CustomWidgetDescriptor>
> }
> ```
> 
> Then strict-check for `null`, truthy or `undefined` card value in `<Widget />` component to decide whether Card should be skipped, overridden or defaulted.
> 
> 1. There are two additional commits that does not belong to this PR

done

**Review by Dergash — APPROVED** [review]

**Pull request merged**